psa-crypto/tests/key_agreement.rs

```diff
@@ -0,0 +1,107 @@
+use psa_crypto::operations::{key_agreement, key_management};
+use psa_crypto::types::algorithm::{KeyAgreement, RawKeyAgreement};
+use psa_crypto::types::key::{Attributes, EccFamily, Lifetime, Policy, Type, UsageFlags};
+use psa_crypto::types::status::Error;
+
+const PEER_PUBLIC_KEY: [u8; 65] = [
+    0x04, 0xd1, 0x2d, 0xfb, 0x52, 0x89, 0xc8, 0xd4, 0xf8, 0x12, 0x08, 0xb7, 0x02, 0x70, 0x39, 0x8c,
+    0x34, 0x22, 0x96, 0x97, 0x0a, 0x0b, 0xcc, 0xb7, 0x4c, 0x73, 0x6f, 0xc7, 0x55, 0x44, 0x94, 0xbf,
+    0x63, 0x56, 0xfb, 0xf3, 0xca, 0x36, 0x6c, 0xc2, 0x3e, 0x81, 0x57, 0x85, 0x4c, 0x13, 0xc5, 0x8d,
+    0x6a, 0xac, 0x23, 0xf0, 0x46, 0xad, 0xa3, 0x0f, 0x83, 0x53, 0xe7, 0x4f, 0x33, 0x03, 0x98, 0x72,
+    0xab,
+];
+
+const OUR_KEY_DATA: [u8; 32] = [
+    0xc8, 0x8f, 0x01, 0xf5, 0x10, 0xd9, 0xac, 0x3f, 0x70, 0xa2, 0x92, 0xda, 0xa2, 0x31, 0x6d, 0xe5,
+    0x44, 0xe9, 0xaa, 0xb8, 0xaf, 0xe8, 0x40, 0x49, 0xc6, 0x2a, 0x9c, 0x57, 0x86, 0x2d, 0x14, 0x33,
+];
+
+const EXPECTED_OUTPUT: [u8; 32] = [
+    0xd6, 0x84, 0x0f, 0x6b, 0x42, 0xf6, 0xed, 0xaf, 0xd1, 0x31, 0x16, 0xe0, 0xe1, 0x25, 0x65, 0x20,
+    0x2f, 0xef, 0x8e, 0x9e, 0xce, 0x7d, 0xce, 0x03, 0x81, 0x24, 0x64, 0xd0, 0x4b, 0x94, 0x42, 0xde,
+];
+
+#[test]
+fn key_agreement() {
+    let alg = RawKeyAgreement::Ecdh;
+    let attributes = Attributes {
+        key_type: Type::EccKeyPair {
```

Comment:
So key agreement can also be done on ECC keys? Maybe we need to update [this code](https://docs.rs/psa-crypto/0.3.0/src/psa_crypto/types/key.rs.html#262) then.
Is it the case that `Ecdh` can be used with ECC keys where `Ffdh` can be used with DH keys?

Reply:
Yes, ECC supports key agreement. Not sure about second question, maybe @ionut-arm has an idea? It also [doesn't look like Mbed Crypto supports FFDH at the moment](https://github.com/ARMmbed/mbedtls/issues/3261#issue-607690687).

Reply:
Yes and yes. I actually asked a similar question back in the day and I think they've included clarification on this in subsequent releases of the spec - found it weird that they didn't explicitly specify that ECC keys are used for `ECDH` when they have a DH key type.

The "classical" Diffie-Hellman (over finite fields) uses DH keys and is represented by `FFDH`, the elliptic-curve variant uses ECC keys and is represented by `ECDH`.

Reply:
Yeah, that code seems wrong, good spot!

Reply:
Cool, thanks for the clarification! In that case, could you update it as part of this PR @sbailey-arm please 😃 ?